src/app.ts

```diff
@@ -64,29 +68,50 @@ async function app() {
       'exit with error if this is not the latest version of SUSHI',
       false
     )
-    .option('-i, --init', 'initialize a SUSHI project')
-    .option('-u, --update-dependencies', 'update FHIR packages in project configuration')
-    .version(getVersion(), '-v, --version', 'print SUSHI version')
+    .action(async function (projectPath, options) {
+      await runBuild(projectPath, options);
+    })
     .on('--help', () => {
       console.log('');
       console.log('Additional information:');
       console.log('  [path-to-fsh-project]');
       console.log('    Default: "."');
       console.log('  -o, --out <out>');
       console.log('    Default: "fsh-generated"');
+    });
+
+  program
+    .command('updateDependencies [path-to-fsh-project]')
+    .description('update FHIR packages in project configuration')
+    .action(async function (projectPath) {
+      const input = ensureInputDir(projectPath);
+      const config: Configuration = readConfig(input);
+      await updateExternalDependencies(config);
+      process.exit(0);
     })
-    .arguments('[path-to-fsh-defs]')
-    .action(function (pathToFshDefs) {
-      input = pathToFshDefs;
-    })
-    .showHelpAfterError()
-    .parse(process.argv)
-    .opts();
+    .on('--help', () => {
+      console.log('');
+      console.log('Additional information:');
+      console.log('  [path-to-fsh-project]');
+      console.log('    Default: "."');
+    });
 
-  if (program.init) {
-    await init();
-    process.exit(0);
+  program
+    .command('init')
+    .description('initialize a SUSHI project')
+    .action(async function () {
+      await init();
+      process.exit(0);
+    });
+
+  // Maintain backwards compatability with prior SUSHI command structure by defaulting to build
+  if (process.argv.length >= 2 && !SUPPORTED_COMMANDS.includes(process.argv[2])) {
+    process.argv = [...process.argv.slice(0, 2), 'build', ...process.argv.slice(2)];
```

Comment:
How thoroughly do we want support backwards compatibility? This won't support the old `sushi --init`.

Reply:
Also, this block makes it so that if you make a typo in one of the commands (or try to use a made-up command), we try to run SUSHI. If we don't have this block, we get an error message that says `error: unknown command 'typo'`. I don't think there's much we can do about this, but it might be worth recognizing this quirk that comes up when trying to be backwards compatible.

On second thought, maybe this block could be smarter and check if the `process.argv.length == 2` or if `process.argv.length > 2 && fs.existsSync(process.argv[2])`. That might catch some of the typo cases, but wouldn't resolve the `sushi --init` backwards compatibility issue.

Reply:
On third thought, we can probably make this even more robust by using the following conditional:

```
if (
  process.argv.length === 2 ||
  (process.argv.length > 2 &&
    !SUPPORTED_COMMANDS.includes(process.argv[2]) &&
    process.argv.slice(2).some(a => fs.existsSync(a)))
)
```

This better supports the fact that options could come before the path to the FSH project.

Reply:
I like your third thought!

Reply:
Yay! Added the third thought in [fe15249](https://github.com/FHIR/sushi/pull/1163/commits/fe15249a27e11d26faf9284e0d4a048f55960078).